src/adapter/src/coord/command_handler.rs

```diff
@@ -469,29 +470,51 @@ impl Coordinator {
             Statement::CreateSource(stmt) => {
                 let internal_cmd_tx = self.internal_cmd_tx.clone();
                 let conn_id = session.conn_id();
-                let purify_fut = mz_sql::pure::purify_create_source(
-                    Box::new(catalog.into_owned()),
-                    self.now(),
-                    stmt,
-                    self.connection_context.clone(),
-                );
+                let now = self.now();
+                let connection_context = self.connection_context.clone();
                 let otel_ctx = OpenTelemetryContext::obtain();
-                task::spawn(|| format!("purify:{conn_id}"), async move {
-                    let result = purify_fut.await.map_err(|e| e.into());
-                    // It is not an error for purification to complete after `internal_cmd_rx` is dropped.
-                    let result = internal_cmd_tx.send(Message::CreateSourceStatementReady(
-                        CreateSourceStatementReady {
-                            session,
-                            tx,
-                            result,
-                            params,
-                            depends_on,
-                            original_stmt,
-                            otel_ctx,
-                        },
-                    ));
-                    if let Err(e) = result {
-                        tracing::warn!("internal_cmd_rx dropped before we could send: {:?}", e);
+
+                task::spawn(|| format!("purify:{conn_id}"), {
+                    let state = catalog.state().clone();
+                    let catalog_conn_id = catalog.conn_id;
+                    let cluster = catalog.cluster;
+                    let database = catalog.database;
+                    let search_path = catalog.search_path;
+                    let role_id = catalog.role_id;
+                    let prepared_statements = catalog.prepared_statements.cloned();
+
+                    async move {
+                        let catalog = ConnCatalog {
```

Comment:
We definitely don't want to spin up ConnCatalogs manually.

Ah, *THIS* is why we needed the Cow: to change the CatalogState from a ref to an owned object in the purify task. `purify_create_source` only uses the ConnCatalog to make StatementContexts which mostly only call `cx.get_item_by_resolved_name(connection)` (the load generator does another thing, but similar). I kinda wonder if we need to split up purify into sync and async parts, where the sync part gets a catalog but the async part doesn't to avoid this.

For this PR: I think we should *keep* the `Cow` and .into_owned so purify doesn't need to change, but keep the rest of the stuff removing mutation of the system vars, along with a new fn that makes a WMR-enabled system conncatalog. That removes the biggest problem of system var mutation while leaving the purify stuff for later work.